### PR TITLE
Clean up of the docs, introducing a deprecated section

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Related to #712 

Deprecates functionality as described in #719. Re-arranges the structure to be cleaner!

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? \[not needed for typos/docs\]
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by akihironitta — APPROVED** [review]
LGTM so far :] Thank you

**Comment by Borda** [comment]
@SeanNaren is this still draft?

**Comment by SeanNaren** [comment]
hey @Borda it's ready to go!

File: docs/source/models_howto.rst [hunk 1]
```diff
@@ -1,405 +0,0 @@
-How to use models
```
Comment (Borda) [inline docs/source/models_howto.rst:1]: shall we keep this one?

Comment (SeanNaren) [reply docs/source/models_howto.rst:1]: Sure this can go into the deprecated section?

**Review by Borda — APPROVED** [review]

**Pull request merged**